File: benchmarks/lua/string/string_concat.lua
-- String concatenation benchmark
-- Concatenate "x" N times using table.concat

local N = tonumber(arg and arg[1]) or 5000000
-- collectgarbage("collect")

local t = {}
for i = 1, N do
    t[i] = "x"
end
local result = table.concat(t)

print(#result)

File: benchmarks/js/string/string_concat.js
// String concatenation benchmark
// Concatenate "x" N times using array push + join

var print = typeof console !== 'undefined' ? console.log.bind(console) : globalThis.print;
var DEFAULT = 5000000;
var N = parseInt(typeof scriptArgs !== 'undefined' ? scriptArgs[1] : (typeof process !== 'undefined' ? process.argv[2] : '0')) || DEFAULT;

var arr = [];
for (var i = 0; i < N; i++) {
    arr.push("x");
}
var result = arr.join("");

print(result.length);
